docs/csharp/language-reference/keywords/file.md

```diff
@@ -0,0 +1,52 @@
+---
+description: "file access modifier: Declare types whose scope is the file in which it's declared"
+title: "file keyword - C# Reference"
+ms.date: 09/15/2022
+f1_keywords: 
+  - "file_CSharpKeyword"
+  - "file"
+helpviewer_keywords: 
+  - "file keyword [C#]"
+---
+# file (C# Reference)
+
+Beginning with C# 11, the `file` contextual keyword is a type access modifier.
+
+File-scoped access restricts a top-level type's scope and visibility to the file in which it's declared. The `file` accessor modifier will generally be applied to types written by a source generator. File scoped types provide source generators with a convenient way to avoid name collisions among generated types. The `file` access modifier declares a file-scoped type, as in this example:
+
+```csharp
+file class HiddenWidget
+{
+    // implementation
+}
+```
+
+Any types nested within a file based type is also only visible within the file in which it's declared. Other types in an assembly may use the same name as a file-scoped type. Because the file-scoped type is visible only in the file where it's declared, these types don't create a naming collision.
```

Comment:
```suggestion
Any types nested within a file-based type are also only visible within the file in which it's declared. Other types in an assembly may use the same name as a file-scoped type. Because the file-scoped type is visible only in the file where it's declared, these types don't create a naming collision.
```